docs/ruleset_functions/get_number_of_floors.md

```diff
@@ -0,0 +1,39 @@
+# get_number_of_floors
+**Schema Version:** 0.0.28  
+
+**Description:** gets the number of floors in the building.  Parking Garages are not counted
+
+**Inputs:**  
+- **RMI**: The baseline ruleset model instance
+
+**Returns:**  
+- **result**: an integer indicating the number of floors
+ 
+**Function Call:**
+- **get_zone_conditioning_category**
+
+## Logic:
+- check to see if Building.number_of_floors_above_grade and Building.number_of_floors_below_grade exisits, if so, the sum of these two numbers will be returned. `if Building.number_of_floors_above_grade != null && Building.number_of_floors_below_grade != null:`
+	- set result equal to the number of floors above grade plus the number of floors below grade: `result = Building.number_of_floors_above_grade + null && Building.number_of_floors_below_grade`
+- otherwise, the number of floors is not explicitly stated, count the number of floors based on floor_name:
+	- use the function get_zone_conditioning_category to get a dictionary of zone id's and their conditioning category - we will only include conditioned zones in the building_area_types_with_total_area_and_zones_dict: `zone_conditioning_category_dict = get_zone_conditioning_category(RMI)`
+	- create a set of the floor names.  We use a set because only unique elements can be added to the set: `floor_names = set()`
+	- loop through the building segments: `for segment in RMI.building.building_segments:`
+		- loop through the zones: `for zone in segment.zones:`
+			- using zone_conditioning_category_dict, check if the zone is conditioned or semi-heated. `if zone_conditioning_category_dict[zone.id] == "CONDITIONED RESIDENTIAL" || zone_conditioning_category_dict[zone.id] == "CONDITIONED NON-RESIDENTIAL" || zone_conditioning_category_dict[zone.id] == "CONDITIONED MIXED" || zone_conditioning_category_dict[zone.id] == "SEMI-HEATED"`:
+				- loop through the spaces in the zone: `for space in zone:`
+					- eligible zones are zones that are not parking garage zones if any of the spaces are not PARKING_GARAGE, then this zone is not exclusively a parking garage space: `if space.lighting_space_type != PARKING_GARAGE:`
+						- add the floor name to the set of floor names.  It doesn't matter if the floor name is already in the set, because a set, by definition can only have one of each item: `floor_names.add(zone.floor_name)`
```

Comment:
@KarenWGard the lighting space type should be 'parking_area_interior' correct?
<img width="380" alt="image" src="https://github.com/pnnl/ruleset-checking-tool/assets/8323384/910cb211-222d-4f4a-a483-af7dea3eb175">

We could also consider checking he zone conditioning category returned through the function, if 'unenclosed' then it would imply a parking garage.

Reply:
updated to parking_area_interior

we area already filtering for any zones that are not one of the acceptable conditioning categories (CONDITIONED... or SEMI-HEATED), so any unenclosed zones will not make it to the line 25 check.